example/App.js

```diff
@@ -24,44 +24,58 @@ export default class App extends Component {
       proMonthlyPrice: "Loading",
       currentID: ""
     };
-    Purchases.addPurchaseListener((productIdentifier, purchaserInfo, error) => {
-      if (error && !error.userCancelled) {
-        this.setState({error: error.message});
-        return;
-      }
-  
-      this.handlePurchaserInfo(purchaserInfo);
-    });
-  
-    Purchases.addPurchaserInfoUpdateListener((purchaserInfo, error) => {
-      if (purchaserInfo) {
-        this.handlePurchaserInfo(purchaserInfo);
-      }
-    });
+  }
   
-    Purchases.addRestoreTransactionsListener((purchaserInfo, error) => {
-      if (purchaserInfo) {
+  setListeners = () => {
+    return {
+      'purchaseListenerID': Purchases.addPurchaseListener((productIdentifier, purchaserInfo, error) => {
+        console.log("addPurchaseListener");
+        if (error && !error.userCancelled) {
+          this.setState({error: error.message});
+          return;
+        }
         this.handlePurchaserInfo(purchaserInfo);
-      }
-    });
+      }),
+      'purchaserInfoUpdatedListenerID': Purchases.addPurchaserInfoUpdateListener((purchaserInfo, error) => {
+        console.log("addPurchaserInfoUpdateListener");
+        if (purchaserInfo) {
+          this.handlePurchaserInfo(purchaserInfo);
+        }
+      }),
+      'restoreTransactionsListenerID': Purchases.addRestoreTransactionsListener((purchaserInfo, error) => {
+        console.log("addRestoreTransactionsListener");
+        if (purchaserInfo) {
+          this.handlePurchaserInfo(purchaserInfo);
+        }
+      })
+    }
   }
 
+
   async componentDidMount() {
     try {
       let purchases = await Purchases.setup("LQmxAoIaaQaHpPiWJJayypBDhIpAZCZN", "purchases_sample_id_4");
       let entitlements = await Purchases.getEntitlements();
       let appUserID = await Purchases.getAppUserID();
+      const listenerIDs = this.setListeners()
```

Comment:
I've never seen this pattern before. 

Is there another example library that uses this?

Reply:
@alfonsoperez suggested that this is common in the Javascript world and I have seen it used here for example Animatedhttps://facebook.github.io/react-native/docs/0.6/animatedvalue#removelistener

But I have also seen things like returning a unsubscribe function that someone can call when unmounting https://rnfirebase.io/docs/v5.x.x/auth/reference/auth#onAuthStateChanged

Or just passing the listener instance https://github.com/Olical/EventEmitter/wiki/EventEmitter.removeListener

I am more used to the latter, but open to do any of the other implementations.

Reply:
Ah, ok great! Just hadn't seen the pattern before.